src/go-slang/lib/heap/index.ts

```diff
@@ -359,6 +372,15 @@ export class Heap {
     this.memory.setUint8(ptr_heap_addr + 3, 0) // initialize buffer size to 0
     return ptr_heap_addr
   }
+  /* Memory Layout of a WaitGroup: 
+   * [0-7:tag][0-7:count] (2 words) 
```

Comment:
Should we just use 1 word instead? We could keep the count in bytes 1:4 (2^32) in the first word instead using a additional word

Reply:
Will do.